Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -239,45 +202,117 @@ void CGPathAddLineToPoint(CGMutablePathRef path, const CGAffineTransform* m, flo
 
     if (pathObj->_count + 1 >= pathObj->_max) {
         pathObj->_max += 32;
-        pathObj->_components = (pathComponent*)IwRealloc(pathObj->_components, pathObj->_max * sizeof(pathComponent));
+        pathObj->_components = (CGPathElement*)IwRealloc(pathObj->_components, pathObj->_max * sizeof(CGPathElement));
     }
 
-    pathObj->_components[pathObj->_count].type = pathComponentLineTo;
-    pathObj->_components[pathObj->_count].point.x = x;
-    pathObj->_components[pathObj->_count].point.y = y;
+    pathObj->_components[pathObj->_count].type = kCGPathElementAddLineToPoint;
+
+    pathObj->_components[pathObj->_count].points = (CGPoint*)IwCalloc(1, sizeof(CGPoint));
```

Comment:
This approach, while really tight on memory usage, is not really buying us a lot and will cause performance issue on two fronts: 

1. Lot of small memory allocations (1, 2, 3 or 4 points).
2. Locality of reference of path element + points.

I suggest that we allocate the buffer long enough to hold CGPathElement and the maximum number of points.  The points buffer will be at the end of the element.  While this may sound bad, it actually isn't since memory wont be fragmented anymore.  Also addresses the other shallow copy comment.